api/middleware.go

```diff
@@ -150,38 +151,19 @@ func (a *API) limitHandler(lmt *limiter.Limiter) middlewareHandler {
 	}
 }
 
-func (a *API) verifyOperatorRequest(w http.ResponseWriter, req *http.Request) (context.Context, error) {
-	c, _, err := a.extractOperatorRequest(w, req)
-	return c, err
-}
-
-func (a *API) extractOperatorRequest(w http.ResponseWriter, req *http.Request) (context.Context, string, error) {
-	token, err := a.extractBearerToken(w, req)
-	if err != nil {
-		return nil, token, err
-	}
-	if token == "" || token != a.config.OperatorToken {
-		return nil, token, unauthorizedError("Request does not include an Operator token")
-	}
-	return withAdminUser(req.Context(), &models.User{ID: uuid.Nil, Email: "operator@netlify.com"}), token, nil
-}
-
 func (a *API) requireAdminCredentials(w http.ResponseWriter, req *http.Request) (context.Context, error) {
-	c, t, err := a.extractOperatorRequest(w, req)
-	if err == nil {
-		return c, nil
-	}
-
-	if t == "" {
+	ctx := req.Context()
+	t, err := a.extractBearerToken(w, req)
+	if err != nil || t == "" {
 		return nil, err
 	}
 
```

Comment:
I think we need to do a `withAdminUser` here to set the admin user to the context. Seems like it was originally being set [here](https://github.com/supabase/gotrue/blob/e3d1d95f7f963041fb1d78e89b9a289e442717ba/api/middleware.go#L166) in `extractOperatorRequest`.

The [adminUserUpdate](https://github.com/supabase/gotrue/blob/e3d1d95f7f963041fb1d78e89b9a289e442717ba/api/admin.go#L95) / Create / Delete methods call `getAdminUser` to retrieve the admin user from the context.

Reply:
my assumption was that in a Supabase context:
1) [`token != a.config.OperatorToken`](https://github.com/supabase/gotrue/blob/e3d1d95f7f963041fb1d78e89b9a289e442717ba/api/middleware.go#L163) would always resolve to true
2) [this error ](https://github.com/supabase/gotrue/blob/e3d1d95f7f963041fb1d78e89b9a289e442717ba/api/middleware.go#L164)was returned
3) and [silently ignored](https://github.com/supabase/gotrue/blob/e3d1d95f7f963041fb1d78e89b9a289e442717ba/api/middleware.go#L171)

i.e. the only way `withAdminUser` could be set, would be if the user passed `OperatorToken` as the bearer token

Reply:
hmm not sure if i missed something but i don't see `withAdminUser` being called anywhere? so we no longer set the adminUser in the context? 

I was thinking we should do the following after line 159
```
ctx = withAdminUser(req.Context(), &models.User{ID: uuid.Nil, Email: "operator@supabase.com"})
```

Reply:
oh i just realised that we set the context [here](https://github.com/supabase/gotrue/blob/1cfdfbaa529d1fba9964c93529f371c88f0f80f5/api/auth.go#L36) if the authentication is valid, so there's no need to do what i suggested above